src/dom-to-image-more.js

```diff
@@ -122,6 +147,34 @@
                 return clone;
             });
         }
+
+        function makeSvgDataUri(node) {
+            let width = options.width || util.width(node);
+            let height = options.height || util.height(node);
+
+            return Promise.resolve(node)
+                .then(function (svg) {
+                    svg.setAttribute('xmlns', 'http://www.w3.org/1999/xhtml');
+                    return new XMLSerializer().serializeToString(svg);
+                })
+                .then(util.escapeXhtml)
+                .then(function (xhtml) {
+                    const foreignObjectSizing =
```

Comment:
It was getting ugly having two `then()` clauses just to format up a nested string. Now we compute everything and build the string up once.